databricks/koalas/internal.py

```diff
@@ -598,6 +600,10 @@ def attach_distributed_sequence_column(sdf, column_name):
         #         ...
         #     }
         sdf = sdf.withColumn(spark_partition_column, F.spark_partition_id())
+
+        # Cache the DataFrame to fix the partition ID.
+        sdf.cache()
```

Comment:
Without this, partition ID could be changed between creating counts and attaching the counts..

Reply:
Also it needs to create cache of the whole dataset, so `head()` or `tail()` will be slow.

Reply:
We might want to add a new default index type using cache here.